octodns/provider/azuredns.py

```diff
@@ -121,6 +120,12 @@ def _params_for_SRV(self, data, key_name, azure_class):
                                       data['value']['target']))
         return {key_name: params}
 
+    def _params_for_TXT(self, data, key_name, azure_class):
+        if 'values' in data:
+            return {key_name: [azure_class([v]) for v in data['values']]}
+        else:  # API for TxtRecord has list of str, even for singleton
+            return {key_name: [azure_class([data['value']])]}
```

Comment:
Might be slightly cleaner to do something like the following to separate the parsing/marshaling bit from where the return value is being built.

```python
try:
    values = data['values']
except KeyError:
    values = [data['value']]
return {key_name: [azure_class([v]) for v in values']}
```